extensions/interactions/TextInput/directives/oppia-interactive-text-input.component.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the TextInput interaction.
+ *
+ * IMPORTANT NOTE: The naming convention for customization args that are passed
+ * into the directive is: the name of the parameter, followed by 'With',
+ * followed by the name of the arg.
+ */
+
+import { ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { TextInputCustomizationArgs } from 'interactions/customization-args-defs';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { InteractionRulesService } from 'pages/exploration-player-page/services/answer-classification.service';
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { Schema } from 'services/schema-default-value.service';
+import { TextInputRulesService } from './text-input-rules.service';
+
+@Component({
+  selector: 'oppia-interactive-text-input',
+  templateUrl: './text-input-interaction.component.html'
+})
+export class InteractiveTextInputComponent implements OnInit {
+  @Input() placeholderWithValue;
+  @Input() rowsWithValue;
+  @Input() savedSolution;
+  @Input() labelForFocusTarget: string;
+  answer: string;
+  placeholder: string;
+  schema: {
+    type: string;
+    'ui_config': {
+      placeholder?: string;
+      rows?: number;
+    };
+  };
```

Comment:
Should we separate this into a type variable?

Reply:
Done!